tests/trestle/core/profile_resolver_test.py

```diff
@@ -236,11 +236,31 @@ def test_replace_params_assignment_mode(simplified_nist_catalog: cat.Catalog) ->
     """Test replacement of params in assignment mode."""
     cat_interface = CatalogInterface(simplified_nist_catalog)
     param_dict = cat_interface._get_full_param_dict()
-    ac_44 = cat_interface.get_control('ac-4.4')
-    ControlInterface.replace_control_prose(ac_44, param_dict, None, ParameterRep.ASSIGNMENT_FORM)
+    ac_44 = copy.deepcopy(cat_interface.get_control('ac-4.4'))
+    orig_prose = ac_44.parts[0].prose
+    # test the case where a selection has parameters in choices but no value was assigned
+    ControlInterface.replace_control_prose(
+        ac_44, param_dict, '[.]', ParameterRep.ASSIGNMENT_FORM, False, 'IBM Assignment:', 'Assignment:'
+    )
+    assert ac_44.parts[
+        0
+    ].prose == 'Prevent encrypted information from bypassing [Assignment: organization-defined information flow control mechanisms] by [Selection (one or more): decrypting the information; blocking the flow of the encrypted information; terminating communications sessions attempting to pass encrypted information;  [IBM Assignment: my procedure] ].'  # noqa E501
+    value = 'blocking the flow of the encrypted information'
+    param_dict['ac-4.4_prm_2'].values = [com.Value(__root__=value)]
```

Comment:
One more comment, I have just ran this test and it failed for me on this line with this error
```
>   ???
E   pydantic.error_wrappers.ValidationError: 1 validation error for Parameter
E   values -> 0 -> __root__
E     str type expected (type=type_error.str)
```

I believe it should be `ParameterValue`?

Reply:
Yes that is weird.  I have fixed it.